Fetch theme editor items from ThemeDB

KoBeWi opened the pull request:
Together with #84760 fixes #82402

KoBeWi committed:
Fetch theme editor items from ThemeDB

YuriSizov commented on editor/plugins/theme_editor_plugin.cpp:
```diff
@@ -2370,18 +2370,25 @@ void ThemeTypeEditor::_update_type_list_debounced() {
 
-		(ThemeDB::get_singleton()->get_default_theme().operator->()->*get_list_func)(default_type, &names);
+		List<ThemeDB::ThemeItemBind> theme_binds;
+		ThemeDB::get_singleton()->get_class_own_items(default_type, &theme_binds);
+		for (const ThemeDB::ThemeItemBind &E : theme_binds) {
+			if (E.data_type == p_type) {
+				names.push_back(E.item_name);
+			}
+		}
```
It's probably worth it to add a new method to filter by the data type and avoid excessive loops.

YuriSizov replied:
Looks like it's needed in https://github.com/godotengine/godot/pull/85015 as well.

YuriSizov reviewed (approved):
Needs a rebase, but otherwise looks good.

Do you think we leave any performance on the table by switching from a method pointer to `get_theme_item_list`? I know this is needed for `ThemeDB` anyway, so it's fine. But I'm not well versed in C++ so curious what your take is here.

YuriSizov reviewed (changes requested):
Actually, as I was testing https://github.com/godotengine/godot/pull/84760, I realized there is something wrong will all 3 PRs. While we can use `get_class_own_items` in the docs (and in fact we should), for the theme editor, the override list, and the argument autocompletion we need to include inherited items as well.

This requires a new `get_class_items` method, which returns not just own items. Plus we need a method to get such items filtered based on the data type, as outlined above. I think one of these PRs need to do all this groundwork, and then others can be built on top of it.

Let me know if you need any help with any of that.

PS. If you need an example of this being a problem, check CheckBox for instance.

KoBeWi commented:
Updated. I added yet another argument to `get_class_items()`, which can be used to filter by type. If I were to add a new method, it would be pretty much the same, so I wanted to avoid code duplication.

akien-mga reviewed (approved):
Code changes look fine to me.

akien-mga commented:
Thanks!
